server/datastore/mysql/testing_utils.go

```diff
@@ -103,10 +103,10 @@ func setupReadReplica(t testing.TB, testName string, ds *Datastore, opts *Datast
 		for _, fk := range fks {
 			stmt := fmt.Sprintf(`ALTER TABLE %s.%s DROP FOREIGN KEY %s`, replicaDB, fk.TableName, fk.ConstraintName)
 			_, err := replica.ExecContext(ctx, stmt)
-			// If the FK was already removed ...
-			if strings.Contains(err.Error(), "check that column/key exists") {
+			if err != nil && strings.Contains(err.Error(), "check that column/key exists") {
```

Comment:
@mna I was getting this weird error on past runs https://github.com/fleetdm/fleet/actions/runs/5525305681/jobs/10078765935?pr=12723#step:9:7129 which seems to be caused by some replication lag? Thoughts?